review-problems.ml

```diff
@@ -676,7 +790,21 @@ module Functions = struct
 
      Rank: **
    *)
-  let poly_gen cs = assert false
+  let poly_gen cs = match cs with
+    | y::ys ->
+      (fun x ->
+         match
+           begin
+             List.fold_left
+               (fun y z -> match y with
+                    e,i -> (Plus (e,Times(Lit z,pow x i)),i+1)
+               )
+               (Lit y,1)
+               ys
+           end
+         with (acc,_) -> acc
+      )
+    | [] -> assert false
```

Comment:
Actually `poly_gen` should work for an empty list of coefficients, and should give you the "zero polynomial", which always returns zero.